src/test/java/book_store/controller/BookControllerTest.java

```diff
@@ -0,0 +1,233 @@
+import java.util.Arrays;
+import java.util.List;
+
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext
+    ) throws SQLException {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/add-three-default-books.sql")
+            );
+        }
+    }
+
+    @AfterAll
+    static void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get all books")
+    void getAll_ReturnsAllBooks_ExpectedSuccess() throws Exception {
+        List<BookDto> expected = new ArrayList<>();
+        expected.add(new BookDto().setId(1L).setTitle("Kobzar").setAuthor("Shevchenko")
+                .setPrice(BigDecimal.valueOf(10.99)).setIsbn("9460303332081"));
+        expected.add(new BookDto().setId(2L).setTitle("Avatar").setAuthor("Unknow")
+                .setPrice(BigDecimal.valueOf(20.99)).setIsbn("9460306342021"));
+        expected.add(new BookDto().setId(3L).setTitle("Terminator").setAuthor("Arnold")
+                .setPrice(BigDecimal.valueOf(30.99)).setIsbn("9460301332081"));
+
+        MvcResult result = mockMvc.perform(
+                        get("/books")
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto[] actual = objectMapper.readValue(result.getResponse().getContentAsByteArray(), BookDto[].class);
+        Assertions.assertEquals(3, actual.length);
+        for (BookDto book : expected) {
+            Assertions.assertTrue(Arrays.stream(actual)
+                    .anyMatch(a -> a.getId().equals(book.getId())
+                            && a.getTitle().equals(book.getTitle())
+                            && a.getAuthor().equals(book.getAuthor())
+                            && a.getPrice().equals(book.getPrice())
+                            && a.getIsbn().equals(book.getIsbn())));
+        }
```

Comment:
containsAll() should work here and will be much shorter (if you have proper equals of course)